content/panorama/scripts/panorama_shop.js

```diff
@@ -489,7 +492,7 @@ function SetItemStock(item, ItemStock) {
 					UpdateSmallItem(child);
 					if (child.BHasClass('CanBuy')) {
 						SendItemBuyOrder(child.itemName);
-						bought = true;
+						bought = child;
```

Comment:
In that case change `var bought = false;` to `var bought;`, so it's type would be `Panel | undefined` instead of `Panel | boolean`